### ISSUE TITLE
Include AsyncMoya and Combine Moya directly into the core dependency

### ISSUE THREAD

**Issue by lordcodes** [Moya/Moya#2262]
Is there a particular reason to have Combine-support and Async-Await support as separate dependencies? I understand RxSwift and ReactiveSwift as they add support for working with 3rd-party reactive libraries. As async-await is a language feature I feel like that can just be baked into the core framework. Combine could be done the same way and wrapped with `#if canImport(Combine)`.

**Comment by gsl-anthonymerle** [comment]
To me, there are 2 reasons on why they did it this way:
 1- (for async/await mostly) because Moya enables support for iOS 10 and later, in which version they aren't available
 2- They seem to have a kind of convention where the very lest of what you need to use Moya is in the Core library, and everything that can be put on top of it to match one's architecture are put on side, optional, extension packages (you don't need async/await or Combine to use Moya, but you need to register their packages to be able to use it in your own app code)

**Comment by lordcodes** [comment]
Yeah I guess, both can be managed via flags to only include them on iOS versions and platforms that support them.

I do understand this argument though.

I would understand not including the Combine Moya now, as I think Swift Concurrency is a better way to do networking. I feel that as async-await is a language feature, you could include support for it directly into the main library and use compiler flags to support older iOS versions. I would expect most libraries would start adding async-await compatible APIs from now on rather than every library offering a separate library to add this, especially when you want to keep the number of dynamic frameworks down.

**Comment by gsl-anthonymerle** [comment]
Yes that's totally understandable 🙏 
Btw, as the AsyncMoya is still not in an official release, can I ask you if you're actually using it on any production code you work on ?
I'm asking because I want to give it a try for a later use, but I'm still hesitating to do so because of its unreleased state

**Comment by lordcodes** [comment]
I have been using it in production for a while, not directly. I based my own implementation on the one from AsyncMoya, however, the code is pretty much the same and works perfectly for me.

**Comment by gsl-anthonymerle** [comment]
Ah! That's the way I decided to go too for now ^^
Thanks for your answers @lordcodes 
I hope the AsyncMoya will be released soon 🤞

**Comment by ERTuringLee** [comment]
Any news about AsyncMoya? Thanks

**Comment by ijunfly** [comment]
Is there any progress?
